service/controller/clusterapi/v17/resources/clusterid/create.go

```diff
@@ -2,8 +2,55 @@ package clusterid
 
 import (
 	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/reconciliationcanceledcontext"
+
+	"github.com/giantswarm/cluster-operator/service/controller/clusterapi/v17/key"
 )
 
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCluster(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring cluster status has cluster ID")
+
+	status := r.commonClusterStatusAccessor.GetCommonClusterStatus(cr)
+
+	if status.ID != "" {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "ensured cluster status has cluster ID")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+
+		return nil
+	}
```

Comment:
I would like the code to be more blocky, but that may only be my personal taste. When I look at the implementation here I see a flat list of code lines which do lots of different things and the scope blocks make it often super clear and easy to read what is going on. You then can also comment on the blocks' header which makes a lot of sense when you look at the files because the code is super structured.

Reply:
Personally I don't think the blocks here would improve so much. This is very simple and I intentionally added empty lines between sections so that they kind of bring up the steps here. This combined with usual Go style where if-section bails out on error and success case continues forward, I feel this is enough. When you think that blocks improve here, push a PR :slightly_smiling_face: